### PR TITLE
Parametrized sem-service

### PR DESCRIPTION
**Pull request by bogyo210**
Ability to add username and password to mysql, postgres, mongodb.
Also provide default socket for mysql and postgres on the host.

### CODE REVIEW

File: sem-service-check-params [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      ;;
+    "mongodb" )
+      shift
+      params-check::mongodb $@
+      ;;
+
+    * )
+      shift
+      echo "$@"
+      ##params error##
```
Comment (shiroyasha) [inline sem-service-check-params]: What happens here? Do we show an error message to the customer, and is the exit status non-zero?

File: sem-service-check-params [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+       --username=* )
+         username=$(echo $words|cut -d'=' -f2)
+         [ "$username" ] && username=" -e MONGO_INITDB_ROOT_USERNAME=$username "
+       ;;
+       --password=* )
+         password=$(echo $words|cut -d'=' -f2)
+         [ "$password" ] && password=" -e MONGO_INITDB_ROOT_PASSWORD=$password "
+       ;;
+       *)
+         ##>&2 echo "Unknown parameter $words"
```
Comment (shiroyasha) [inline sem-service-check-params]: What happens here? Do we show an error message to the customer, and is the exit status non-zero?

**Comment by shiroyasha** [comment]
It looks great! I've left several questions about the implementation.

Additionally, there is a big number of commits on this Pull Request. Please rebase and squash the reaccuring commits. @CommanderK5 can help with this.

File: sem-service-check-params (hunk 2 above)
Comment (bogyo210) [reply sem-service-check-params]: No error message, the garbage is just overlooked when starting the container.

**Comment by bogyo210** [comment]
commits squashed to 1

File: sem-service-check-params [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+       *)
+       ;;
+      esac
+    done
+    default_params="$default_params$username$password "
+  fi
+  echo "$service_version $default_params"
+}
+
+params-check::mongodb () {
```
Comment (shiroyasha) [inline sem-service-check-params]: Please check this.

Comment (bogyo210) [reply sem-service-check-params]: Removed duplicate function.

**Pull request merged**